src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp

```diff
@@ -1399,7 +1403,9 @@ void PDPServer::process_changes_release_(
             }
             else
             {
-                if (!edp->process_and_release_change(ch))
+                bool ret = (discovery_db_.is_writer(ch) || discovery_db_.is_reader(ch));
+
+                if (!ret || (ret && !edp->process_and_release_change(ch)))
```

Comment:
Operator `||` has short-circuit mechanisms, so no need to check `ret` again:
```suggestion
                if (!ret || !edp->process_and_release_change(ch, false))
```